v0/destinations/hs/transform.js

```diff
@@ -324,6 +326,8 @@ const processRouterDest = async inputs => {
   const errorRespList = [];
   // using the first destination config for transforming the batch
   const { destination } = inputs[0];
+  // reduce the no. of calls for properties endpoint
+  const propertyMap = await getProperties(destination);
```

Comment:
Properties are needed if at least one of the events has `traits` defined. So, let's further optimize it.

Reply:
added the `traits` check